### PR TITLE
Add component usage example in the MDX docs

### PR DESCRIPTION
**Pull request by MadLittleMods**
#### What kind of changes does this PR include?

- New or updated content

#### Description

Add Astro and React component usage example in the MDX docs. This was pretty hard to figure out even though the docs say it's possible. And there is confusion with the old `setup: |` syntax that is [spread](https://github.com/snowpackjs/astro/issues/491) [across](https://craigglennie.com/posts/2021/rewriting-in-astro) the internet already.

The only component usage example I could find in the existing docs comes from the migration guide converting an old React `.md` to `.mdx` in https://docs.astro.build/en/migrate/#converting-existing-md-files-to-mdx

Relevant docs:

 - https://docs.astro.build/en/guides/markdown-content/#using-components-in-mdx
    - This PR adds an example here
 - https://docs.astro.build/en/guides/integrations-guide/mdx/#components
    - It would be nice to cross-link or have an example here too (suggestion welcome)
    - Update: it looks like we sorta have an example in https://docs.astro.build/en/guides/integrations-guide/mdx/#custom-components but this is a little different for overriding existing components. And I completely missed this when Googling and trying to figure this out.

Update: Or perhaps link to https://github.com/withastro/astro/blob/latest/examples/with-mdx/src/pages/index.mdx?plain=1 which I just found from https://docs.astro.build/en/guides/integrations-guide/mdx/#examples

### CODE REVIEW

File: src/pages/en/guides/markdown-content.md [hunk 1]
```diff
@@ -273,6 +273,22 @@ With the `@astrojs/mdx` integration, you can use Astro or UI framework component
 
 Don't forget to include a `client:directive` if necessary!
 
+```astro title="src/pages/about.mdx"
```
Comment (MadLittleMods) [inline src/pages/en/guides/markdown-content.md:276]: The syntax here might need updating. I wasn't sure what to put for the `{3,6}` stuff, etc

Comment (delucis) [reply src/pages/en/guides/markdown-content.md:276]: I’d suggest this here:

````suggestion
```astro title="src/pages/about.mdx" {5-6} /<.+\/>/
````

- `{5-6}` will highlight lines 5 to 6 where the import statements are.
- `/<.+\/>/` is regular expression that will match both `<Button />` and `<ReactCounter />` and highlight the usage.

**Review by delucis — COMMENTED** [review]
Thanks @MadLittleMods! I’ve left a suggestion for the code block syntax.

File: src/pages/en/guides/markdown-content.md (hunk 1 above)
Comment (MadLittleMods) [reply src/pages/en/guides/markdown-content.md]: Thanks for the review and explanation @delucis ❤️

**Review by delucis — APPROVED** [review]
Thanks again @MadLittleMods! Welcome to team docs 🥳

**Comment by MadLittleMods** [comment]
Thanks for the review and merge @delucis 👯

**Pull request merged**